event_model/__init__.py

```diff
@@ -2423,6 +2424,7 @@ def __call__(
             time=time,
             uid=uid,
             hints=hints,
+            **kwargs,
```

Comment:
If we do the `doc.update(kwargs)` I mentioned above then maybe we should write a check that the metadata doesn't have any keys matching the required or optional ones in the TypedDict.

Reply:
Additionally could we change `compose_run` to create a `RunStart` directly and add the metadata afterwards? I probably should have done this in my initial conversion to TypedDicts